examples/mongocxx/mongodb.com/documentation_examples.cpp

```diff
@@ -1180,6 +1193,85 @@ void delete_examples(mongocxx::database db) {
     }
 }
 
+static void snapshot_examples(mongocxx::client& client) {
+    // Write concern "majority" needed to avoid the following error below:
+    //
+    // Unable to read from a snapshot due to pending collection catalog changes;
+    // please retry the operation.
```

Comment:
Was this error observed when running this example?

This may suggest there are other conditions where this error appears.

SERVER-41532 includes "insert in a transaction" in the Steps to Reproduce. This example does not start a transaction.
SERVER-41532 notes this applies to sharded clusters. This example is only run against a replica set.

Reply:
I observed this error from this example in CI. More context is below:
```
[2022/11/18 20:29:37.878] terminate called after throwing an instance of 'mongocxx::v_noabi::query_exception'
[2022/11/18 20:29:37.878]   what():  Unable to read from a snapshot due to pending collection catalog changes; please retry the operation. Snapshot timestamp is Timestamp(1668803377, 209). Collection minimum is Timestamp(1668803377, 210): generic server error
[2022/11/18 20:29:37.884] bash: line 200: 431523 Aborted                 (core dumped) $test
[2022/11/18 20:29:37.884] Command ''shell.exec' in "test"' failed: shell script encountered problem: exit code 134.
```
I was able to work around it by adding a short sleep between inserting to the new collection and starting the snapshot, but setting write concern to majority also solved this.

Reply:
Out of curiosity, does this behavior surprise you? Do you think this could be a bug?

Reply:
The error suggests it is a transient. The Python implementation includes a [helper to wait for the snapshot to become available](https://github.com/mongodb/mongo-python-driver/commit/e325b24b78e431cb889c5902d00b8f4af2c700c3#diff-c5d782e261f04fca18024ab18c3ed38fb45ede24cde4f9092e012f6fcbbe0df5R1368). I suggest adding that helper.